app/src/main/java/org/thoughtcrime/securesms/onboarding/LandingActivity.kt

```diff
@@ -19,28 +53,128 @@ class LandingActivity : BaseActionBarActivity() {
+            Spacer(modifier = Modifier.weight(1f))
+
+            OutlineButton(
+                text = stringResource(R.string.onboardingAccountCreate),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                contentDescription = GetString(R.string.AccessibilityId_create_account_button)
+            ) { startPickDisplayNameActivity() }
+            Spacer(modifier = Modifier.height(14.dp))
+            FilledButton(
+                text = stringResource(R.string.onboardingAccountExists),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                contentDescription = GetString(R.string.AccessibilityId_restore_account_button)
+            ) { startLinkDeviceActivity() }
+            Spacer(modifier = Modifier.height(8.dp))
+            BorderlessButton(
+                text = stringResource(R.string.onboardingTosPrivacy),
+                modifier = Modifier
+                    .width(262.dp)
+                    .align(Alignment.CenterHorizontally),
+                contentDescription = GetString(R.string.AccessibilityId_open_url),
+                fontSize = 11.sp,
+                lineHeight = 13.sp
+            ) { openDialog() }
+            Spacer(modifier = Modifier.height(8.dp))
+        }
+    }
+
+    private fun openDialog() {
+        showSessionDialog {
+            title(R.string.urlOpen)
+            text(R.string.urlOpenBrowser)
+            button(
+                R.string.activity_landing_terms_of_service,
+                contentDescriptionRes = R.string.AccessibilityId_terms_of_service_link
+            ) { open("https://getsession.org/terms-of-service") }
+            button(
+                R.string.activity_landing_privacy_policy,
+                contentDescriptionRes = R.string.AccessibilityId_privacy_policy_link
+            ) { open("https://getsession.org/privacy-policy") }
+        }
+    }
+
+    private fun open(url: String) {
+        Intent(Intent.ACTION_VIEW, Uri.parse(url)).let(::startActivity)
+    }
+
+    @Composable
+    private fun IncomingText(text: String) {
+        ChatText(
+            text,
+            color = classicDarkColors[2]
+        )
+    }
+
+    @Composable
+    private fun ColumnScope.OutgoingText(text: String) {
+        ChatText(
+            text,
+            color = session_accent,
+            textColor = MaterialTheme.colors.primary,
+            modifier = Modifier.align(Alignment.End)
+        )
     }
 
-    private fun link() {
-        val intent = Intent(this, LinkDeviceActivity::class.java)
-        push(intent)
+    @Composable
+    private fun ChatText(
+        text: String,
+        color: Color,
+        modifier: Modifier = Modifier,
+        textColor: Color = Color.Unspecified
+    ) {
+        Text(
+            text,
+            fontSize = 16.sp,
```

Comment:
This is the same issue here where text values are hardcoded. We should be following the designs files, with a limited set of predetermined text styles, and define those styles in a manner that can be reused throughout the app.

Reply:
part of onboarding